src/userSync.js

```diff
@@ -0,0 +1,175 @@
+import * as utils from 'src/utils';
+
+export function newUserSync() {
+  // Set user sync config default values which can be overridden by the publisher
+  const userSyncDefaultConfig = {
+    syncEnabled: true,
+    pixelEnabled: true,
+    syncsPerBidder: 5
+  }
+
+  let publicApi = {};
+  // A queue of user syncs for each adapter
+  // Let getDefaultQueue() set the defaults
+  let queue = getDefaultQueue();
+
+  // Since user syncs require cookie access we want to prevent sending syncs if cookies are not supported
+  let cookiesAreSupported = !utils.isSafariBrowser() && utils.cookiesAreEnabled();
```

Comment:
do we want to block this or leave it to the caller? Talking about safari specifically.